### PR TITLE
ci: upload test result XML files to CircleCI after tests

### PR DESCRIPTION
**Pull request by josephperrott**
Uploading test results to CircleCI will provide both the test failure information in the CircleCI UI
as well as allowing CircleCI to provide insights information about test success rates on an individual
test level.

This is useful both in this repository, but will also act as a proving ground for making this kind of
information available in other repositories we manage to gather better metrics.

[Here](https://app.circleci.com/pipelines/github/angular/dev-infra/415/workflows/abe37406-8cb7-43db-8974-feda2aab5323/jobs/841/tests) is an example of what failing tests look like in the CircleCI UI.

### CODE REVIEW

File: .circleci/copy-junit-reports.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      if (file === 'test.xml') {
+        files.push(filePath);
+      }
+    }
+  }
+  return files;
+};
+
+/** StdOut result from the bazel info command. */
+const bazelInfoOutput = execSync('yarn bazel info', {stdio: 'pipe', encoding: 'utf8'});
```
Comment (devversion) [inline .circleci/copy-junit-reports.js:31]: bazel info bazel-testlogs
```suggestion
const bazelInfoOutput = execSync('yarn bazel info bazel-testlogs', {stdio: 'pipe', encoding: 'utf8'});
```

File: .circleci/copy-junit-reports.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * Discover all test results, which @bazel/jasmine stores as `test.xml` files, in the directory and
+ * return back the list of absolute file paths.
+ */
+const findTestResultsInDir = function (dirPath, files) {
+  for (const file of readdirSync(dirPath)) {
+    const filePath = join(dirPath, file);
+    if (statSync(filePath).isDirectory()) {
+      files = findTestResultsInDir(filePath, files);
+    } else {
+      if (file === 'test.xml') {
```
Comment (devversion) [inline .circleci/copy-junit-reports.js:22]: It seems like Bazel does not guarantee this being named like that. Can we use all
XML files? Just an optional comment/question. It does not matter a lot

**Review by devversion — APPROVED** [review]

File: .circleci/copy-junit-reports.js (hunk 1 above)
Comment (josephperrott) [reply .circleci/copy-junit-reports.js:31]: I googled this for like 10 minutes and could not find how to do this.  I feel silly now.

File: .circleci/copy-junit-reports.js (hunk 2 above)
Comment (josephperrott) [reply .circleci/copy-junit-reports.js:22]: Sure we can just use all XML files, as it looks like they are the only files.

File: .circleci/copy-junit-reports.js [hunk 3]
```diff
@@ -19,18 +19,20 @@ const findTestResultsInDir = function (dirPath, files) {
...
     }
   }
   return files;
 };
 
-/** StdOut result from the bazel info command. */
-const bazelInfoOutput = execSync('yarn bazel info', {stdio: 'pipe', encoding: 'utf8'});
 /** Absolute path to the bazel instance's testlog directory.  */
-const testLogPath = bazelInfoOutput.match(/bazel\-testlogs: (.*)/)[1];
+const testLogPath = execSync('yarn bazel info bazel-testlogs', {
```
Comment (devversion) [inline .circleci/copy-junit-reports.js:32]: Nit: Should we use --silent for Yarn or will that pollute the stdout?

**Review by devversion — APPROVED** [review]
LGTM, one minor nit

File: .circleci/copy-junit-reports.js (hunk 3 above)
Comment (josephperrott) [reply .circleci/copy-junit-reports.js]: Good call, I have it set to always be silent in my env, so I forget about it sometimes.

**Pull request closed without merging**